src/http/request_context.rs

```diff
@@ -5,28 +5,29 @@ use hyper::HeaderMap;
 
 use super::{DefaultHttpClient, Response, ServerContext};
 use crate::blueprint::Server;
-use crate::config;
+use crate::config::{self, Upstream};
 
 #[derive(Setters)]
 pub struct RequestContext {
   pub http_client: DefaultHttpClient,
   pub server: Server,
```

Comment:
do we even need `server`? the idea of moving upstream out was to actually make the client decoupled with the server settings.

Reply:
I tried to remove the server field from RequestContext. But there are some values needed in the Server.
e.g. 
https://github.com/tailcallhq/tailcall/blob/main/src/lambda/expression.rs#L124
https://github.com/tailcallhq/tailcall/blob/main/src/lambda/evaluation_context.rs#L61

What do you think? Move these fields upstream. I'm just afraid to incur extra conflict.

Reply:
I think we can do this in a separate issue. Would you mind creating one?

Reply:
sure! #504